examples/example_glfw_wgpu/CMakeLists.txt

```diff
@@ -13,105 +12,60 @@
+)
+include(${CMAKE_CURRENT_BINARY_DIR}/cmake/CPM.cmake)
 
-set(CMAKE_CXX_STANDARD 17) # Dawn requires C++17
+CPMAddPackage(
+  NAME dawn
+  GITHUB_REPOSITORY google/dawn
+  VERSION 20250607.112309
+  OPTIONS
+    "DAWN_FETCH_DEPENDENCIES ON"
+  EXCLUDE_FROM_ALL YES
+  GIT_SHALLOW TRUE
+  GIT_SUBMODULES "docs" # docs is not a submodules but is used to prevent submodule resolution
+)
 
-# Dear ImGui
 set(IMGUI_DIR ../../)
-
-# Libraries
-if(EMSCRIPTEN)
-  if(EMSCRIPTEN_VERSION VERSION_GREATER_EQUAL "3.1.57")
-    set(IMGUI_EMSCRIPTEN_GLFW3 "--use-port=contrib.glfw3" CACHE STRING "Choose between --use-port=contrib.glfw3 and -sUSE_GLFW=3 for GLFW implementation (default to --use-port=contrib.glfw3)")
-  else()
-    # cannot use contrib.glfw3 prior to 3.1.57
-    set(IMGUI_EMSCRIPTEN_GLFW3 "-sUSE_GLFW=3" CACHE STRING "Use -sUSE_GLFW=3 for GLFW implementation" FORCE)
-  endif()
-  set(LIBRARIES glfw)
-  add_compile_options(-sDISABLE_EXCEPTION_CATCHING=1 -DIMGUI_DISABLE_FILE_FUNCTIONS=1)
-else()
-  # Dawn wgpu desktop
-  set(DAWN_FETCH_DEPENDENCIES ON)
-  set(IMGUI_DAWN_DIR CACHE PATH "Path to Dawn repository")
-  if (NOT IMGUI_DAWN_DIR)
-    message(FATAL_ERROR "Please specify the Dawn repository by setting IMGUI_DAWN_DIR")
-  endif()
-
-  option(DAWN_FETCH_DEPENDENCIES "Use fetch_dawn_dependencies.py as an alternative to using depot_tools" ON)
-
-  # Dawn builds many things by default - disable things we don't need
-  option(DAWN_BUILD_SAMPLES "Enables building Dawn's samples" OFF)
-  option(TINT_BUILD_CMD_TOOLS "Build the Tint command line tools" OFF)
-  option(TINT_BUILD_DOCS "Build documentation" OFF)
-  option(TINT_BUILD_TESTS "Build tests" OFF)
-  if (NOT APPLE)
-    option(TINT_BUILD_MSL_WRITER "Build the MSL output writer" OFF)
-  endif()
-  if(WIN32)
-    option(TINT_BUILD_SPV_READER "Build the SPIR-V input reader" OFF)
-    option(TINT_BUILD_WGSL_READER "Build the WGSL input reader" ON)
-    option(TINT_BUILD_GLSL_WRITER "Build the GLSL output writer" OFF)
-    option(TINT_BUILD_GLSL_VALIDATOR "Build the GLSL output validator" OFF)
-    option(TINT_BUILD_SPV_WRITER "Build the SPIR-V output writer" OFF)
-    option(TINT_BUILD_WGSL_WRITER "Build the WGSL output writer" ON)
-  endif()
-
-  add_subdirectory("${IMGUI_DAWN_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/dawn" EXCLUDE_FROM_ALL)
-
-  set(LIBRARIES webgpu_dawn webgpu_cpp webgpu_glfw glfw)
-endif()
-
 add_executable(example_glfw_wgpu
   main.cpp
-  # backend files
   ${IMGUI_DIR}/backends/imgui_impl_glfw.cpp
   ${IMGUI_DIR}/backends/imgui_impl_wgpu.cpp
-  # Dear ImGui files
   ${IMGUI_DIR}/imgui.cpp
   ${IMGUI_DIR}/imgui_draw.cpp
   ${IMGUI_DIR}/imgui_demo.cpp
   ${IMGUI_DIR}/imgui_tables.cpp
   ${IMGUI_DIR}/imgui_widgets.cpp
 )
-IF(NOT EMSCRIPTEN)
-  target_compile_definitions(example_glfw_wgpu PUBLIC
-    "IMGUI_IMPL_WEBGPU_BACKEND_DAWN"
-  )
-endif()
+target_compile_definitions(example_glfw_wgpu PUBLIC
+  "IMGUI_IMPL_WEBGPU_BACKEND_DAWN"
```

Comment:
I'm not sure if removing this entirely is the direction to go. (and removing it from backends/imgui_impl_wgpu.cpp) so this would be the default. (Perhaps best left to a finalized webgpu api in any case)